tests/base/dataset_test.py

```diff
@@ -1049,3 +1049,33 @@ def test_cat_features_warning(iris, caplog):
     Dataset(iris, cat_features=[])
     assert_that(caplog.records, has_length(1))
 
+def test_dataset_duplicate_column_names_validation(iris: pd.DataFrame):
+    """
+    Function checks whether the Dataset object's duplicate column name
+    validation raises `DeepchecksValueError` when the data has duplicate column names.
+    """
+
+    # Duplicate column names
+    columns = ["sepal length (cm)", "sepal width (cm)", "sepal length (cm)",
+               "sepal width (cm)", "target"]
+    
+    duplicated_columns = ["sepal length (cm)", "sepal width (cm)"]
+
+    # data now has duplicate column names
+    iris.columns = columns
```

Comment:
The tests are failing because you're modifying the iris fixture in place. Please use a copy of the iris dataset `iris_copy = iris.copy()` and modify it's columns instead.

Reply:
> Great, approved once the test issues and pylint issues are resolved.

Sure, I'll work on these changes.